packages/block-editor/src/components/page-template-picker/picker.js

```diff
@@ -29,9 +31,26 @@ const __experimentalPageTemplatePicker = ( { templates = getDefaultTemplates() }
 			<Preview
 				template={ templatePreview }
 				onDismiss={ () => setTemplatePreview( undefined ) }
+				onApply={ () => {
+					setLayout( templatePreview );
+					setTemplatePreview( undefined );
```

Comment:
I wonder if we can group this two actions somehow, since this is generating two undo steps. So when you apply a template and undo, it will retain the title until you undo again.